Build: allow dynamically change docker image

markelog opened the pull request:
Right now, only for couple DBs.
With the 2.1, you can do `${TAG:-5.6}` without needing the `.env` file, but
it seems it would be better for back-compat to support `.env` right now.

Fixes #17891

markelog committed:
Build: allow dynamically change docker image

Right now, only for couple DBs.
With the 2.1, you can do `${TAG:-5.6}` without needing the `.env` file, but
it seems it would be better for back-compat to support `.env` right now.

Fixes #17891

marefr commented:
@markelog as you can see in issue #17891 @papagian is already working on this. Please sync with her how to merge your work.

papagian commented:
@marefr my implementation for #17891 is almost similar to this one, but it didn't have time to test it throughly and therefore I hadn't create a PR. So I guess we can keep @markelog's.
I will review it ASAP!

markelog committed:
Build: add prefix for removal process

papagian commented on Makefile:
```diff
@@ -125,7 +125,7 @@ devenv: devenv-down
 	@cd devenv; \
 	./create_docker_compose.sh $(targets) || \
-	(rm -rf docker-compose.yaml; exit 1)
+	(rm -rf {docker-compose.yaml,conf.tmp,.env}; exit 1)
```
Is `conf.tmp` related with this issue? If not, I would proposed to be removed from this pull request and create a separate pull request for this one.

markelog replied:
Yeah, it is related, some of the blocks do use it and `create_docker_compose.sh` utilize it as well. 

I didn't know about it before tho

papagian reviewed (changes requested):
Nice!
Maybe you could add somewhere some guidelines for overriding the default versions (5.6 for mysql and 9.3) by passing the `TAG` to the Makefile for example:
`make devenv sources=mysql TAG=5.7.18` & `make devenv sources=postgres TAG=9.2`

markelog commented:
> Nice!
Maybe you could add somewhere some guidelines for overriding the default versions (5.6 for mysql and 9.3) by passing the TAG to the Makefile for example:
make devenv sources=mysql TAG=5.7.18 & make devenv sources=postgres TAG=9.2

Yeah, I have separate issue about that - https://github.com/grafana/grafana/issues/16107. In that document we should definitely mention the `TAG` thing

markelog committed:
Docs: mention makefile command instead of the bash

Since we are moving to makefile for things and make command
simplifies some of the actions you have to perform to work with
different environments

markelog committed:
Docs: mention dynamic tags in the docs

markelog committed:
Docs: avoid name collision

Fixes #16107

markelog commented:
It turns out we already some docs for #16107 so I just mentioned them in there. Also rewrote the thing, so there wouldn't be any name collisions and naming is more explicit now

markelog committed:
Merge branch 'master' of github.com:grafana/grafana into env_block_dbs

* 'master' of github.com:grafana/grafana:
  grafana/toolkit: update the way config is being passed to jest cli (#18115)
  Build: detect changes to packages based on the git diff (#18118)
  Build: Release packages under next tag when changes detected on master (#18062)
  SQLStore: allow to look for `is_disabled` flag (#18032)

torkelo reviewed (approved):
Looks great
